src/async-filter.js

```diff
@@ -22,6 +22,17 @@
                         scope.$applyAsync(); // Automatic safe apply, if scope provided
                     }
                 });
+
+                if (scope && scope.$on) {
+                    // Clean up subscription and its last value when the scope is destroyed.
+                    scope.$on('$destroy', () => {
+                        if (subscriptions[inputId]) {
+                            subscriptions[inputId].dispose();
```

Comment:
Need to check if the subscription has `dispose` method since this filter also works with Promises, which don't have a dispose method.